src/ocean/task/Task.d

```diff
@@ -462,7 +462,16 @@ public abstract class Task : ISuspendable
             Task.task_pending_cleanup = this;
             this.fiber.yieldAndThrow(e);
             Task.task_pending_cleanup = null;
-            return true;
+
+            // if task was resumed with `kill` from 'pending cleanup' state
+            // just quit the method as if it was originally terminated by kill
+            if (this.to_kill)
+            {
+                debug_trace("<{}> termination (killed)", cast(void*) this);
```

Comment:
Just to check: the `to_kill` flag should remain set to true, in this case?

Reply:
Yes, but looks like my other fix to reset flags in `assignTo` method was only merged into later branch. I will add it here too.

Reply:
Updated